src/Files.App/Utils/Git/GitHelpers.cs

```diff
@@ -62,6 +66,13 @@ private set
 
 		public static event EventHandler? GitFetchCompleted;
 
+		public static void TryDispose()
+		{
+			_owningThread?.Dispose();
+			_owningThread = null;
+			_activeOperationsCount = 0;
```

Comment:
```suggestion
			var threadToDispose = _owningThread;
			_owningThread = null;
			Interlocked.Exchange(ref _activeOperationsCount, 0);
			threadToDispose?.Dispose();
```
`Dispose()` may take some time so we should null `_owningThread` first.